salesagility/SuiteCRM
# SugarEmail: Fix "to" field not being filled when the last record doesn't have an email

## Description

In the list view when selecting multiple records or a page and the last element doesn't have
and email then nothing gets listed in the "to" field of the email widget.
Similarly when multiple are selected and some don't have an email then invalid addresses (<>)
are listed in the "to" field.

This was introduced in #5640 (likely by accident, see discussion there) by checking the last
email instead of during looping over them when deciding when to add one.

This moves the empty email check into the loop.

## Motivation and Context

We have some records without emails and in case they end up in a selection then emails are broken.

## How To Test This

* Go to a list view
* Select an entry with an email
* Select an entry below without an email
* Action menu > Email
* The email of the first entry should be in the "to" field

## Types of changes

### Final checklist
